### PR TITLE
bulk_indexer: Remove ES field mapper field preview

### PR DESCRIPTION
**Pull request by kruskall**
Removes the field mapper exception field preview returned in the error reason by Elasticsearch.

### CODE REVIEW

File: bulk_indexer.go [hunk 1]
```diff
@@ -139,11 +139,17 @@ func init() {
...
 										// failed to parse field [%s] of type [%s] in %s. Preview of field's value: '%s'
 										// https://github.com/elastic/elasticsearch/blob/588eabe185ad319c0268a13480465966cef058cd/server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java#L234
 										item.Error.Reason, _, _ = strings.Cut(
-											i.ReadString(), ". Preview",
+											reason, ". Preview",
+										)
+										// Match Elasticsearch field mapper field value:
+										// Non-standard token 'NaN': enable `JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS` to allow\n at [Source: (byte[])%s
+										item.Error.Reason, _, _ = strings.Cut(
+											item.Error.Reason, ": enable `JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS` to allow",
```
Comment (carsonip) [inline bulk_indexer.go:152]: This seems very targeted. WDYT about just removing reason when error.type is `x_content_parse_exception`?

**Review by 1pkg — DISMISSED** [review]

**Review by simitt — CHANGES_REQUESTED** [review]
++ on Carson's comment in https://github.com/elastic/go-docappender/pull/190#discussion_r1693073045

**Review by carsonip — COMMENTED** [review]
I'm still investigating the 2 different "types" / "reason" of error under x_content_parse_exception. Let's keep this PR open until we have a better idea of what's happening.

File: bulk_indexer.go [hunk 2]
```diff
@@ -160,6 +161,11 @@ func init() {
 							item.Error.Reason = ""
 						}
 
+						// For unavailable_shards_exception, remove item.Error.Reason
```
Comment (carsonip) [inline bulk_indexer.go:164]: nit: wrong comment.
nit: this if statement can be merged into the previous if statement for simplicity
```suggestion
						// For x_content_parse_exception, remove item.Error.Reason
```

**Review by carsonip — APPROVED** [review]
lgtm, thanks

**Pull request merged**